src/components/DocumentSelector/index.js

```diff
@@ -3,56 +3,55 @@ import * as React from 'react'
 import { h, Component } from 'preact'
 import style from './style.css'
 import { kebabCase } from '../utils/string'
-import { find } from '../utils/object'
+import { isEmpty } from '../utils/object'
 import classNames from 'classnames'
-
-type DocumentTypeOption = {
-  eStatementAccepted?: boolean,
-  warning?: string,
-  hint?: string,
-  icon: string,
-  label: string,
-  value: string,
-}
+import { idDocumentOptions, poaDocumentOptions } from './documentTypes'
+import type { DocumentOptionsType, GroupType } from './documentTypes'
 
 type Props = {
   className?: string,
-  documentTypes: { [string]: any },
+  documentTypes: Object,
+  country?: string,
   i18n: Object,
   setDocumentType: string => void,
   nextStep: () => void,
 }
 
 type WithDefaultOptions = {
-  defaultOptions: Object => DocumentTypeOption[],
+  defaultOptions: Object => DocumentOptionsType[],
 }
 
+const defaultCountryAvailability: ?string => boolean = () => true
+
 // The value of these options must match the API document types.
 // See https://documentation.onfido.com/#document-types
 class DocumentSelector extends Component<Props & WithDefaultOptions> {
 
   getOptions = () => {
-    const {i18n, documentTypes, defaultOptions} = this.props
+    const {i18n, documentTypes, defaultOptions, country = 'GBR' } = this.props
     const defaultDocOptions = defaultOptions(i18n)
-    
-    const options = defaultDocOptions.filter((option) => documentTypes && documentTypes[option.value])
+
+    const options = defaultDocOptions
+      .filter(({ isAvailableInCountry = defaultCountryAvailability }) => isAvailableInCountry(country))
+      .filter(({ value }) => isEmpty(documentTypes) || documentTypes[value])
```

Comment:
I'm struggling to read this. Is there a way of make this a bit more readable?

Reply:
Sure!